csharp/src/Ice/Communicator-Properties.cs

```diff
@@ -52,6 +52,65 @@ internal PropertyValue(string v, bool u)
         public bool? GetPropertyAsBool(string name) =>
             GetPropertyAsInt(name) is int intValue ? intValue > 0 : (bool?)null;
 
+        public int? GetPropertyAsByteSize(string name)
+        {
+            lock (_properties)
+            {
+                if (_properties.TryGetValue(name, out PropertyValue? pv))
+                {
+                    pv.Used = true;
+
+                    if (pv.Val == "unlimited")
+                    {
+                        return 0;
+                    }
+
+                    // Match an integer followed letters
+                    Match match = Regex.Match(pv.Val, @"^([0-9]+[\.]?[0-9]*)([A-Z]+)$");
```

Comment:
This will also match `9.G`. Maybe something like:

```suggestion
                     Match match = Regex.Match(pv.Val, @"^([0-9]+(?:\.[0-9]+)?)([A-Z]+)$");
```